### DISCUSSION TITLE
Access modifier of default plugins not matches on android and iOS

### DISCUSSION THREAD

**Discussion by no-dap** [ionic-team/capacitor#4024] [General] [answered]
I was trying to create some new plugins which extend the default plugin. (PushNotifications class in my case)
It was able to inherit the default plugin's class as the parent of my new plugin witch have an access modifier `public` in android,  
but couldn't inherit in iOS because the class has a `public` access modifier instead of `open`.  
Is there any reason that the default plugin's classes and its methods can't be `open` instead of `public` in swift?

**Comment by imhoffd** [comment]
Are you referring to `CAPPlugin`: https://github.com/ionic-team/capacitor/blob/2.x/ios/Capacitor/Capacitor/CAPPlugin.m? 

Which methods in particular are you trying to access?

**Reply by no-dap** [reply]
I'm trying to inherit the `CAPPushNotificationsPlugin` class: https://github.com/ionic-team/capacitor/blob/2.x/ios/Capacitor/Capacitor/Plugins/PushNotifications.swift#L12
and its public method `load()` in my custom plugin.  

but all plugins in this directory are not inheritable.

**Reply by no-dap** [reply]
I think it's slightly different but the `public` access modifier works similar outside of the module if we finalize the class in java, which is not :  
https://github.com/ionic-team/capacitor/blob/2.x/android/capacitor/src/main/java/com/getcapacitor/plugin/PushNotifications.java#L35

**Reply by no-dap** [reply]
It was my bad. I had changed the word `base plugin` to `default plugin`

**Comment by jcesarmobile** [comment, marked as answer]
The Capacitor plugins are not intended to be subclassed.

For Capacitor 3 all plugins are being moved out of capacitor/ios and capacitor/android to be their own packages, most of them will have "implementation" classes that can be used from native or from other plugins, directly without instantiating the plugin class.
